feat: add 1.239.1 release notes (#14486)
newrelic/docs-website: src/content/docs/release-notes/new-relic-browser-release-notes/browser-agent-release-notes/browser-agent-v1.239.1.mdx

---
subject: Browser agent

> releaseDate: "2023-09-01”

Reviewer: i think this might need to be a normal `"` instead of slanted

Reviewer 2: Yup, that was it! Thanks @roadlittledawn

> 
> 
> 
> 
> ---

Reviewer 3: Remove empty lines.

```suggestion
---
```

bugs: ["Fix issue with leaking event listeners","Issues with types resolution", "Fix internal release process"]
security: []
---

## v1.239.1

### Features

#### Add best effort to detect Ajax events before instantiation
Add a "best effort" at detecting ajax events that may have happened on the page before the agent was instantiated by utilizing the ResourceTiming browser API.

#### Add mode to enable agent to not harvest until user consent
Add ability for customers to wait to start harvesting. This behavior will utilize a configuration setting in tandem with an API call to trigger features to begin harvesting.

#### Remove allow_bfcache flag
Removes unused code which still relied on the window unload event. Going forward, the only method the agent will use to monitor potential end of page session will be visibility change in accordance with modern web practice.

#### Remove old supportability metrics
Removing supportability metrics that are no longer needed including polyfill detection, bfcache usage, body and query bytes sent count, and custom attributes size.

### Bug Fixes

#### Fix issue with leaking event listeners

> Send buffered session trace data in certain cases when it's being discarded at the tail end of a captured session.

Reviewer 3: ```suggestion
Fixing an issue where session timeout would result in rewrapping global event handlers. For instances where the site would be open for long periods of time, such as kiosks, this would eventually result in a degradation in event handler performance.
```

> #### Fix distributed tracing handling of empty string fetch parameter
> Fixing an issue where session timeout would result in rewrapping global event handlers. For instances where the site would be open for long periods of time, such as kiosks, this would eventually result in a degradation in event handler performance.
> 

Reviewer 3: This was from the previous version.

```suggestion
```

#### Issues with types resolution
Fixing issues with typescript throwing errors when trying to resolve the typings for individual features with "moduleResolution": "node" set in the tsconfig.json. "moduleResolution": "nodenext" and "moduleResolution": "node16" will enable typescript use of the exports field of our package.json but could also break other libraries and forces developers to change all local imports to include file extensions.

> #### Fix internal release process
> Fixing issues internal build pipeline failing to deploy new code changes.
> 

Reviewer 3: Not sure we really need to include this one.

```suggestion
```

## Support statement:

New Relic recommends that you upgrade the agent regularly to ensure that you're getting the latest features and performance benefits. Older releases will no longer be supported when they reach [end-of-life](https://docs.newrelic.com/docs/browser/browser-monitoring/getting-started/browser-agent-eol-policy/). Release dates are reflective of the original publish date of the agent version.

New Browser Agent releases are rolled out to customers in small stages over a period of time. Because of this, the date the release becomes accessible to your account may not match the original publish date. Please see this [status dashboard](https://newrelic.github.io/newrelic-browser-agent-release/) for more information.

> Consistent with our [browser support policy](https://docs.newrelic.com/docs/browser/new-relic-browser/getting-started/compatibility-requirements-browser-monitoring/#browser-types), v1.239.1 of the Browser agent was built for and tested against these browsers and version ranges: Chrome 106-114, Edge 106-113, Safari 15-16, and Firefox 107-113. For mobile devices, v1.238.0 was built and tested for Android Chrome 100 and iOS Safari 15.2-16.2.

Reviewer 3: Update browser versions

```suggestion
Consistent with our [browser support policy](https://docs.newrelic.com/docs/browser/new-relic-browser/getting-started/compatibility-requirements-browser-monitoring/#browser-types), v1.239.1 of the Browser agent was built for and tested against these browsers and version ranges: Chrome 107-116, Edge 107-115, Safari 15-16, and Firefox 108-116. For mobile devices, v1.238.0 was built and tested for Android Chrome 100 and iOS Safari 15.2-16.2.
```
